lib/glimesh/account_follows.ex

```diff
@@ -18,49 +18,52 @@ defmodule Glimesh.AccountFollows do
 
   defp sub_and_return(topic), do: {Phoenix.PubSub.subscribe(Glimesh.PubSub, topic), topic}
 
-  defp broadcast({:error, _reason} = error, _event), do: error
-
-  defp broadcast({:ok, %Follower{} = following}, :followers = event) do
+  defp broadcast(%Follower{} = following, :followers = event) do
     Glimesh.Events.broadcast(
       get_subscribe_topic(:follows, following.streamer_id),
       get_subscribe_topic(:follows),
       event,
       following
     )
-
-    {:ok, following}
   end
 
   def follow(%User{} = streamer, %User{} = user, live_notifications \\ false) do
-    attrs = %{
-      has_live_notifications: live_notifications
-    }
 
-    results =
+    result =
       %Follower{
         streamer: streamer,
-        user: user
+        user: user,
+        has_live_notifications: live_notifications
       }
-      |> Follower.changeset(attrs)
+      |> Follower.changeset()
       |> Repo.insert()
 
-    channel = ChannelLookups.get_channel_for_user(streamer)
+    with {:ok, follower} <- result
```

Comment:
This `with` is to gracefully handle when `followChannel` mutation is invoked but the user is already following the streamer, previously the insert would through a constraint violation exception, now it jumps to the else and returns an error tuple that absinthe can display nicely.